ZupIT/ritchie-cli
# split local repository by workspace

### Description

##### Local repos
Now when we create a formula it will be built and sent to a specific repository according to the name of your workspace with the prefix **"local"**, for example:
if we have a workspace with the name my-work-name.

```go
Workspace {
  Name: "my-work-name",
  Dir: "/home/user/my-workspace-path"
}
```
we will have a repository called "local-my-work-name" that will have the highest priority.

Remembering that with this modification we gained some commands for local repositories such as `rit list repo`, `rit set repo-priority`, and `rit delete repo`

---
##### Repo priority
A problem with the repository priority has been fixed, now when setting a repository with negative priority (-1) this repo will have the highest priority having its priority set to 0. When setting a repo with a priority higher than the amount of repository you have added that will have the lowest priority. for example, if we have a list with 5 repositories and you set the priority of some to 99 this will have priority 4 where this is the lowest priority among the repositories.

##### Local build modifications

Today, by default, if a formula doesn't have the /bin folder, it means it hasn't been built yet, so Ritchie builds it automatically when the user types the formula command. The logic for executing build scripts with Makefile, build.sh, or build.bat has been removed from the local build to keep in one place, so we won the build with docker for local formulas.

### How to verify it
Build this PR and create formulas locally and execute the `rit list repo` command to see your repositories.

### Changelog
split local repository by workspace
